### PR TITLE
javadoc for utils

### PR DESCRIPTION
**Pull request by JkSelf**
A subtask of #91
Complete the javadoc for utils.

### CODE REVIEW

File: src/main/java/com/intel/chimera/utils/IOUtils.java [hunk 1]
```diff
@@ -24,6 +24,16 @@
 import org.apache.commons.logging.Log;
 
 public class IOUtils {
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/IOUtils.java]: Please also add javadoc for `IOUtils ` class

File: src/main/java/com/intel/chimera/utils/OSInfo.java [hunk 2]
```diff
@@ -27,11 +27,34 @@
 public class OSInfo {
   private static HashMap<String, String> archMapping = new HashMap<String, String>();
 
+  /**
+   * The string for X86.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/OSInfo.java]: /**
 * The constant string represents for X86 architecture, the value is: {@value #X86}.
 */

File: src/main/java/com/intel/chimera/utils/OSInfo.java [hunk 3]
```diff
@@ -27,11 +27,34 @@
...
+   */
   public static final String X86 = "x86";
+
+  /**
+   * The string for X86_64.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/OSInfo.java]: The constant string represents for X86_64 architecture, the value is: {@value #X86_64}. 

Others are same.

File: src/main/java/com/intel/chimera/utils/OSInfo.java [hunk 4]
```diff
@@ -86,14 +109,29 @@ public static void main(String[] args) {
...
+   * Gets the OS name.
+   *
+   * @return return the OS name.
+   */
   public static String getOSName() {
     return translateOSNameToFolderName(System.getProperty("os.name"));
   }
 
+  /**
+   * Gets the arch name.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/OSInfo.java]: arch -> architecture

File: src/main/java/com/intel/chimera/utils/OSInfo.java [hunk 5]
```diff
@@ -86,14 +109,29 @@ public static void main(String[] args) {
...
+   *
+   * @return return the arch name.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/OSInfo.java]: arch -> architecture

File: src/main/java/com/intel/chimera/utils/OSInfo.java [hunk 6]
```diff
@@ -86,14 +109,29 @@ public static void main(String[] args) {
...
+   * @return return the current OS's native lib folder.
+   */
   public static String getNativeLibFolderPathForCurrentOS() {
     return getOSName() + "/" + getArchName();
   }
 
+  /**
+   * Gets the OS name.
+   *
+   * @return return the OS name.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/OSInfo.java]: (One return)

File: src/main/java/com/intel/chimera/utils/OSInfo.java [hunk 7]
```diff
@@ -86,14 +109,29 @@ public static void main(String[] args) {
     System.out.print(getNativeLibFolderPathForCurrentOS());
   }
 
+  /**
+   * Gets the native lib folder.
+   *
+   * @return return the current OS's native lib folder.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/OSInfo.java]: (One return)

File: src/main/java/com/intel/chimera/utils/OSInfo.java [hunk 8]
```diff
@@ -140,6 +184,12 @@ static String translateOSNameToFolderName(String osName) {
     }
   }
 
+  /**
+   * Translates the archname to folder name.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/OSInfo.java]: architecture name

File: src/main/java/com/intel/chimera/utils/OSInfo.java [hunk 9]
```diff
@@ -140,6 +184,12 @@ static String translateOSNameToFolderName(String osName) {
...
+   *
+   * @param archName the archname.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/OSInfo.java]: @param archName the architecture name.

File: src/main/java/com/intel/chimera/utils/ReflectionUtils.java [hunk 10]
```diff
@@ -72,14 +83,11 @@
...
    * If no such property is specified, then <code>defaultValue</code> is 
-   * returned.
-   * 
-   * An exception is thrown if the returned class does not implement the named
-   * interface. 
+   * returned.An exception is thrown if the returned class does not
+   * implement the named interface.
    * 
    * @param name the class name.
    * @param defaultValue default value.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/ReflectionUtils.java]: the class name of default implementation.

File: src/main/java/com/intel/chimera/utils/Utils.java [hunk 11]
```diff
@@ -89,7 +89,11 @@ private static void loadChimeraSystemProperties() {
     }
   }
 
-  /** Forcibly free the direct buffer. */
+   /**
+   * Forcibly free the direct buffer.
+    *
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/Utils.java]: remove a blank space

File: src/main/java/com/intel/chimera/utils/Utils.java [hunk 12]
```diff
@@ -113,18 +123,39 @@ public static int getBufferSize(Properties props) {
     }
   }
 
+  /**
+   * Gets the cipher class.
+   *
+   * @param props The <code>Properties</code> class represents a set of
+   *              properties.
+   * @return return the cipher class based on the props.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/Utils.java]: one return

File: src/main/java/com/intel/chimera/utils/Utils.java [hunk 13]
```diff
@@ -113,18 +123,39 @@ public static int getBufferSize(Properties props) {
...
     return props.getProperty(configName) != null ? props.getProperty(configName) : System
         .getProperty(configName, CHIMERA_CRYPTO_CIPHER_CLASSES_DEFAULT);
   }
 
+  /**
+   * Gets the Jce provider.
+   *
+   * @param props The <code>Properties</code> class represents a set of
+   *              properties.
+   * @return return the jce provider based on the props.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/Utils.java]: one return

File: src/main/java/com/intel/chimera/utils/Utils.java [hunk 14]
```diff
@@ -113,18 +123,39 @@ public static int getBufferSize(Properties props) {
...
         props.getProperty(CHIMERA_CRYPTO_CIPHER_JCE_PROVIDER_KEY) :
         System.getProperty(CHIMERA_CRYPTO_CIPHER_JCE_PROVIDER_KEY);
   }
 
+  /**
+   * Gets the random device path.
+   *
+   * @param props The <code>Properties</code> class represents a set of
+   *              properties.
+   * @return return the random device path based on the props.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/Utils.java]: one return

File: src/main/java/com/intel/chimera/utils/Utils.java [hunk 15]
```diff
@@ -136,27 +167,53 @@ public static String getRandomDevPath(Properties props) {
     return devPath;
   }
 
+  /**
+   * Gets the lib path.
+   *
+   * @return return the lib path.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/Utils.java]: one return

File: src/main/java/com/intel/chimera/utils/Utils.java [hunk 16]
```diff
@@ -136,27 +167,53 @@ public static String getRandomDevPath(Properties props) {
     return devPath;
   }
 
+  /**
+   * Gets the lib path.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/Utils.java]: Gets the path of native library.

File: src/main/java/com/intel/chimera/utils/Utils.java [hunk 17]
```diff
@@ -136,27 +167,53 @@ public static String getRandomDevPath(Properties props) {
...
+   * @return return the lib path.
+   */
   public static String getLibPath() {
     return System.getProperty(CHIMERA_CRYPTO_LIB_PATH_KEY);
   }
 
+  /**
+   * Gets the lib name.
+   *
+   * @return return the lib name.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/Utils.java]: one return

File: src/main/java/com/intel/chimera/utils/Utils.java [hunk 18]
```diff
@@ -136,27 +167,53 @@ public static String getRandomDevPath(Properties props) {
...
+   * Gets the lib path.
+   *
+   * @return return the lib path.
+   */
   public static String getLibPath() {
     return System.getProperty(CHIMERA_CRYPTO_LIB_PATH_KEY);
   }
 
+  /**
+   * Gets the lib name.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/Utils.java]: Gets the file name of native library.

File: src/main/java/com/intel/chimera/utils/Utils.java [hunk 19]
```diff
@@ -136,27 +167,53 @@ public static String getRandomDevPath(Properties props) {
...
+   * @return return the lib name.
+   */
   public static String getLibName() {
     return System.getProperty(CHIMERA_CRYPTO_LIB_NAME_KEY);
   }
 
+  /**
+   * Gets the temp directory.
+   *
+   * @return return the temp directory.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/Utils.java]: one name

File: src/main/java/com/intel/chimera/utils/Utils.java [hunk 20]
```diff
@@ -136,27 +167,53 @@ public static String getRandomDevPath(Properties props) {
...
+   * Gets the lib name.
+   *
+   * @return return the lib name.
+   */
   public static String getLibName() {
     return System.getProperty(CHIMERA_CRYPTO_LIB_NAME_KEY);
   }
 
+  /**
+   * Gets the temp directory.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/Utils.java]: Gets the temp directory for extracting crypto library

File: src/main/java/com/intel/chimera/utils/Utils.java [hunk 21]
```diff
@@ -136,27 +167,53 @@ public static String getRandomDevPath(Properties props) {
...
+   * @return return the temp directory.
+   */
   public static String getTmpDir() {
     return System.getProperty(CHIMERA_CRYPTO_LIB_TEMPDIR_KEY,
         System.getProperty("java.io.tmpdir"));
   }
 
-  /** AES/CTR/NoPadding is required */
+  /**
+   * Checks whether the cipher is AES/CTR/NoPadding.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/Utils.java]: Checks whether the cipher is supported streaming.

File: src/main/java/com/intel/chimera/utils/Utils.java [hunk 22]
```diff
@@ -201,7 +258,12 @@ public static void calculateIV(byte[] initIV, long counter, byte[] IV) {
   }
 
   /**
-   * Helper method to create a Cipher instance and throws only IOException
+   * Helper method to create a Cipher instance and throws only IOException.
+   *
+   * @param props The <code>Properties</code> class represents a set of
+   *              properties.
+   * @param transformation the CipherTransformation instance.
+   * @throws IOException if an I/O error occurs.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/Utils.java]: @return

File: src/main/java/com/intel/chimera/utils/Utils.java [hunk 23]
```diff
@@ -268,13 +332,13 @@ public static void checkState(boolean expression) {
...
+   * Splits class names sequence into substrings, Trim each substring into an
    * entry,and returns an list of the entries.
    *
    * @param clazzNames a string consist of a list of the entries joined by a
-   *                   delimiter
-   * @param separator  a delimiter for the input string
-   * @return a list of entries
+   *                   delimiter.
+   * @param separator  a delimiter for the input string.
+   * @return a list of entries.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/utils/Utils.java]: a list of class entries

**Comment by sundapeng** [comment]
Thank @JkSelf for your contribution, I left some comments.

**Comment by JkSelf** [comment]
Thanks for @sundapeng comments. I have fixed.

**Comment by sundapeng** [comment]
+1

**Comment by sundapeng** [comment]
Thank @JkSelf , merged to master

**Pull request merged**